src/bosh_azure_cpi/spec/unit/disk_manager2_spec.rb

```diff
@@ -313,6 +313,150 @@
+            'use_root_disk' => false
+          },
+          'root_disk' => {
+            'size' => 8129
+          }
+        )
+      end
+
+      it 'should return the size of the root disk' do
+        expect(
+          disk_manager2.ephemeral_os_disk(vm_name, stemcell_info, vm_props.root_disk.size, vm_props.ephemeral_disk.size, vm_props.ephemeral_disk.use_root_disk, vm_props.root_disk.placement)
+        ).to eq(
+          disk_name: disk_name,
+          disk_size: 8,
+          disk_placement: nil
+        )
+      end
+    end
+
+    context 'when use_root_disk is false and placement is resource-disk' do
+      let(:vm_props) do
+        props_factory.parse_vm_props(
+          'instance_type' => 'STANDARD_A1',
+          'ephemeral_disk' => {
+            'size' => 2048,
+            'use_root_disk' => false
+          },
+          'root_disk' => {
+            'size' => 3069,
+            'placement' => 'resource-disk'
+          }
+        )
+      end
+
+      it 'should return the size of the root disk' do
+        expect(
+          disk_manager2.ephemeral_os_disk(vm_name, stemcell_info, vm_props.root_disk.size, vm_props.ephemeral_disk.size, vm_props.ephemeral_disk.use_root_disk, vm_props.root_disk.placement)
+        ).to eq(
+          disk_name: disk_name,
+          disk_size: 3,
+          disk_placement: 'ResourceDisk'
+        )
+      end
+    end
+
+    context 'when use_root_disk is true and placement is default' do
+      let(:vm_props) do
+        props_factory.parse_vm_props(
+          'instance_type' => 'STANDARD_A1',
+          'ephemeral_disk' => {
+            'size' => 2048,
+            'use_root_disk' => true
+          },
+          'root_disk' => {
+            'size' => 3069
+          }
+        )
+      end
+
+      it 'should return nil for placement' do
+        expect(
+          disk_manager2.ephemeral_os_disk(vm_name, stemcell_info, vm_props.root_disk.size, vm_props.ephemeral_disk.size, vm_props.ephemeral_disk.use_root_disk, vm_props.root_disk.placement)
+        ).to eq(
+          disk_name: disk_name,
+          disk_size: 5,
+          disk_placement: nil
+        )
+      end
+    end
+
+    context 'when use_root_disk is true and placement is cache-disk' do
+      let(:vm_props) do
+        props_factory.parse_vm_props(
+          'instance_type' => 'STANDARD_A1',
+          'ephemeral_disk' => {
+            'size' => 2048,
+            'use_root_disk' => true
+          },
+          'root_disk' => {
+            'placement' => 'cache-disk'
+          }
+        )
+      end
+
+      it 'should return the size of the root disk' do
+        expect(
+          disk_manager2.ephemeral_os_disk(vm_name, stemcell_info, vm_props.root_disk.size, vm_props.ephemeral_disk.size, vm_props.ephemeral_disk.use_root_disk, vm_props.root_disk.placement)
+        ).to eq(
+          disk_name: disk_name,
+          disk_size: 5,
+          disk_placement: 'CacheDisk'
+        )
+      end
```

Comment:
The test case states that it should return the size of the root disk, but it returns the size of the root disk + the size of the ephemeral disk. Or do I miss something here?